nflow-engine/src/main/java/io/nflow/engine/service/WorkflowDefinitionService.java

```diff
@@ -96,4 +113,29 @@ public void addWorkflowDefinition(WorkflowDefinition wd) {
     }
     logger.info("Added workflow type: {} ({})", wd.getType(), wd.getClass().getName());
   }
+
+  /**
+   *
+   */
+  private synchronized void refreshStoredDefinitions() {
+    long now = currentTimeMillis();
+    if (nextCheckOfStoredDefinitions > now) {
+      return;
+    }
+    nextCheckOfStoredDefinitions = now + MINUTES.toMillis(1);
```

Comment:
make the period configurable, one minute is quite short?

Reply:
maybe replace the boolean config parameter with the period config, and disable if period is configured to < 1

Reply:
The only ways to trigger the load are:
* listing all workflow types - which is only called by the rest api and thus most likely only by the explorer ui on page load
* if someone is trying to insert a workflow type that is not already known, which should be extremely rare

I was thinking that waiting one minute after deploying code until the new workflow types can be used is a long time period. But making it configurable would be good anyway